src/test/java/com/atlassian/db/replica/api/ThrottledCacheTest.java

```diff
@@ -0,0 +1,75 @@
+package com.atlassian.db.replica.api;
+
+import org.junit.Test;
+
+import java.util.ConcurrentModificationException;
+import java.util.Random;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatNoException;
+
+public class ThrottledCacheTest {
+    private final ExecutorService executor = Executors.newFixedThreadPool(10);
```

Comment:
why 10 ?

Reply:
I'm lacking test with  spawning 10 threads and testing concurrent blocking on `lock.tryLock()`

Reply:
Maybe unbounded?

Reply:
You could add a test with 10 requests, without timeout and CountDownLatch in between.